### PR TITLE
Diffable WMS

### PR DESCRIPTION
**Pull request by na9da**
Merge after #4814 

### What this PR does

Fixes https://github.com/TerriaJS/terriajs/issues/4667
Moves the custom code for diffing WMS catalog item to a new catalog item `DiffableWebMapServiceCatalogItem`.

Testing:
Use a catalog item like [this one](https://gist.github.com/na9da/12e8c0dd6de431748228b459c1a5f0ea) with `type` set to `diffable-wms`.

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/Models/DiffableWebMapServiceCatalogItem.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  }
+}
+
+/**
+ * A custom WMS catalog item that supports diffing imagery from two dates.
+ */
+export default class DiffableWebMapServiceCatalogItem extends DiffableMixin(
+  ExtendModel(WebMapServiceCatalogItem, DiffableWebMapServiceCatalogItemTraits)
+) {
+  static readonly type = "diffable-wms";
```
Comment (rowanwins) [inline lib/Models/DiffableWebMapServiceCatalogItem.ts:86]: So does this mean if adding from a catalog json that you need to do something like 

````
{
   "type": "diffable-wms",
   "url": "http://something",
   "name": "my diffable wms"
}
````

File: lib/Models/DiffableWebMapServiceCatalogItem.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+  @computed
+  get disableDateTimeSelector() {
+    return this.catalogItem.isShowingDiff;
+  }
+
+  @computed
+  get shortReport() {
+    const catalogItem = this.catalogItem;
+    console.log("**here**", catalogItem.isShowingDiff);
```
Comment (rowanwins) [inline lib/Models/DiffableWebMapServiceCatalogItem.ts]: stray console

File: lib/Models/DiffableWebMapServiceCatalogItem.ts (hunk 1 above)
Comment (na9da) [reply lib/Models/DiffableWebMapServiceCatalogItem.ts:86]: Yeah that's right. Sorry I didn't make that clear :grimacing:

**Comment by soyarsauce** [comment]
based on extendmodel, but redo with mixins

**Pull request closed without merging**